### PR TITLE
mgr/dashboard: cheroot moved into a separate project 

### PR DESCRIPTION
**Pull request by jschmid1**
CherryPy > 9.0.0 moved the cheroot module into a separate project. See here https://github.com/cherrypy/cherrypy/pull/1548

Introduced in https://github.com/ceph/ceph/pull/31014

Signed-off-by: Joshua Schmid <jschmid@suse.de>

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tspmelo** [comment]
jenkins test dashboard

**Comment by tspmelo** [comment]
Please use `mgr/dashboard` as the component name in the commit.

**Comment by tspmelo** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/cherrypy_backports.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def patch_builtin_ssl_wrap(v, new_wrap):
     if v < StrictVersion("9.0.0"):
         from cherrypy.wsgiserver.ssl_builtin import BuiltinSSLAdapter as builtin_ssl
     else:
-        from cherrypy.cheroot.ssl.builtin import BuiltinSSLAdapter as builtin_ssl
```
Comment (sebastian-philipp) [inline src/pybind/mgr/dashboard/cherrypy_backports.py]: Hm. Was there a timeframe, where `cherook` was already separate (extracted from wsgiserver), but still was part of `cherrypy`?

Comment (jschmid1) [reply src/pybind/mgr/dashboard/cherrypy_backports.py]: looking at the version diff of v8.9.1..v9.0.0 (there are no versions in-between), I can see the hard cut for cheroot.
While `BuiltinSSLAdapter` is not present in 9.0.0, it still is in 8.9.1.

**Comment by callithea** [comment]
jenkins test dashboard

File: src/pybind/mgr/dashboard/cherrypy_backports.py (hunk 1 above)
Comment (votdev) [reply src/pybind/mgr/dashboard/cherrypy_backports.py]: > `cherook`

I assume you're working too much with rook. 😂

**Pull request merged**